packages/main/src/plugin/kubernetes-client.ts

```diff
@@ -1049,64 +1010,159 @@ export class KubernetesClient {
+
+        delete spec.metadata.annotations['kubectl.kubernetes.io/last-applied-configuration'];
+        spec.metadata.annotations['kubectl.kubernetes.io/last-applied-configuration'] = JSON.stringify(spec);
+
+        if (!spec.metadata.namespace) {
+          spec.metadata.namespace = namespace || DEFAULT_NAMESPACE;
         }
-        const groupVersion = this.groupAndVersion(manifest.apiVersion);
-        const namespaceToUse = manifest.metadata?.namespace || optionalNamespace || DEFAULT_NAMESPACE;
-        if (groupVersion.group === '') {
-          const client = ctx.makeApiClient(CoreV1Api);
-          await this.createV1Resource(client, manifest, namespaceToUse);
-        } else if (groupVersion.group === 'apps') {
-          const k8sAppsApi = this.kubeConfig.makeApiClient(AppsV1Api);
-          if (manifest.kind === 'Deployment') {
-            await k8sAppsApi.createNamespacedDeployment(namespaceToUse, manifest);
-          } else if (manifest.kind === 'DaemonSet') {
-            await k8sAppsApi.createNamespacedDaemonSet(namespaceToUse, manifest);
+        try {
+          // try to get the resource, if it does not exist an error will be thrown and we will
+          // end up in the catch block
+          await client.read(spec);
+          // we got the resource, so it exists: patch it
+          //
+          // Note that this could fail if the spec refers to a custom resource. For custom resources
+          // you may need to specify a different patch merge strategy in the content-type header
+          //
+          // See: https://github.com/kubernetes/kubernetes/issues/97423
+          if (action === 'apply') {
+            const response = await client.patch(spec);
+            created.push(response.body);
           }
-        } else if (groupVersion.group === 'networking.k8s.io') {
-          // Add networking object support (Ingress for now)
-          const k8sNetworkingApi = this.kubeConfig.makeApiClient(NetworkingV1Api);
-          if (manifest.kind === 'Ingress') {
-            await k8sNetworkingApi.createNamespacedIngress(namespaceToUse, manifest);
-          }
-        } else {
-          const client = ctx.makeApiClient(CustomObjectsApi);
-          const apiResource = await this.getAPIResource(client, groupVersion, manifest.kind);
-          await this.createCustomResource(
-            client,
-            groupVersion.group,
-            groupVersion.version,
-            apiResource.name,
-            apiResource.namespaced ? namespaceToUse : undefined,
-            manifest,
-          );
+        } catch (error) {
+          // we did not get the resource, so it does not exist: create it
+          const response = await client.create(spec);
+          created.push(response.body);
         }
       }
-      return undefined;
-    } catch (error) {
-      telemetryOptions = { error: error };
+      return created;
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    } catch (error: any) {
```

Comment:
do not use any but unknown

Reply:
Changed, but this led to cludgy any code for checking response that I'm not sure how to deal with.